### ISSUE TITLE
React 0.13 Broken for 0.8.x

### ISSUE THREAD

**Issue by robcolburn** [akiran/react-slick#165]
Just happened over the weekend all versions of 0.8.x break my automated tests that touch React Slick in a React 0.13 project. Downgraded to 0.7.2 to avoid the pain.

```
  Error: Invariant Violation: Slider.render(): A valid ReactComponent must be returned. You may have returned undefined, an array or some other invalid object.
      at invariant (node_modules/react/lib/invariant.js:42:15)
      at [object Object].ReactCompositeComponentMixin._renderValidatedComponent (node_modules/react/lib/ReactCompositeComponent.js:821:46)
      at [object Object].wrapper [as _renderValidatedComponent] (node_modules/react/lib/ReactPerf.js:70:21)
      at [object Object].ReactCompositeComponentMixin.mountComponent (node_modules/react/lib/ReactCompositeComponent.js:237:30)
      at [object Object].wrapper [as mountComponent] (node_modules/react/lib/ReactPerf.js:70:21)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:38:35)
      at ReactDOMComponent.ReactMultiChild.Mixin.mountChildren (node_modules/react/lib/ReactMultiChild.js:192:44)
      at ReactDOMComponent.Mixin._createContentMarkup (node_modules/react/lib/ReactDOMComponent.js:289:32)
      at ReactDOMComponent.Mixin.mountComponent (node_modules/react/lib/ReactDOMComponent.js:199:12)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:38:35)
      at [object Object].ReactCompositeComponentMixin.mountComponent (node_modules/react/lib/ReactCompositeComponent.js:247:34)
      at [object Object].wrapper [as mountComponent] (node_modules/react/lib/ReactPerf.js:70:21)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:38:35)
      at [object Object].ReactCompositeComponentMixin.mountComponent (node_modules/react/lib/ReactCompositeComponent.js:247:34)
      at [object Object].wrapper [as mountComponent] (node_modules/react/lib/ReactPerf.js:70:21)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:38:35)
      at mountComponentIntoNode (node_modules/react/lib/ReactMount.js:248:32)
      at ReactReconcileTransaction.Mixin.perform (node_modules/react/lib/Transaction.js:134:20)
      at batchedMountComponentIntoNode (node_modules/react/lib/ReactMount.js:269:15)
      at ReactDefaultBatchingStrategyTransaction.Mixin.perform (node_modules/react/lib/Transaction.js:134:20)
      at Object.ReactDefaultBatchingStrategy.batchedUpdates (node_modules/react/lib/ReactDefaultBatchingStrategy.js:66:19)
      at Object.batchedUpdates (node_modules/react/lib/ReactUpdates.js:110:20)
      at Object.ReactMount._renderNewRootComponent (node_modules/react/lib/ReactMount.js:404:18)
      at Object.wrapper [as _renderNewRootComponent] (node_modules/react/lib/ReactPerf.js:70:21)
      at Object.ReactMount.render (node_modules/react/lib/ReactMount.js:493:32)
      at Object.wrapper [as render] (node_modules/react/lib/ReactPerf.js:70:21)
      at Object.ReactTestUtils.renderIntoDocument (node_modules/react/lib/ReactTestUtils.js:52:18)
      at Context.<anonymous> (test/DynamicLead/index.js:10:33)
```

**Comment by jordansexton** [comment]
+1

I was running `"react-slick": "0.8.2"`, so something happened that
- introduced a breaking change in a patch version of what is claimed to be known-good @ https://www.npmjs.com/package/react-slick
- overwrote an existing semver release on NPM

How did this even happen?

**Comment by robcolburn** [comment]
@jordansexton,
So, NPM versions cannot be overwritten, but… in s module's package.json you can loosely specify the modules the this particular module depend on. Npm provides the ability to say, "this version plus any patch versions after it" (`~`), or even "this version or any minor versions after it" (`^`). When you install a version'd module, npm grabs the latest applicable dependencies with it. And, this can lead to breakage. The way to avoid that, is with `npm shrinkwrap`, which you should be doing for anything "in production". It's  annoying to maintain a shrinkwrap file, but it's the price you pay for stability.

**Comment by akiran** [comment]
can you post your error messages here

**Comment by jordansexton** [comment]
Mine is identical to @robcolburn's OP:
```
 Error: Invariant Violation: Slider.render(): A valid ReactComponent must be returned. You may have returned undefined, an array or some other invalid object.
      at invariant (node_modules/react/lib/invariant.js:42:15)
      at [object Object].ReactCompositeComponentMixin._renderValidatedComponent (node_modules/react/lib/ReactCompositeComponent.js:821:46)
      at [object Object].wrapper [as _renderValidatedComponent] (node_modules/react/lib/ReactPerf.js:70:21)
      at [object Object].ReactCompositeComponentMixin.mountComponent (node_modules/react/lib/ReactCompositeComponent.js:237:30)
      at [object Object].wrapper [as mountComponent] (node_modules/react/lib/ReactPerf.js:70:21)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:38:35)
      at ReactDOMComponent.ReactMultiChild.Mixin.mountChildren (node_modules/react/lib/ReactMultiChild.js:192:44)
      at ReactDOMComponent.Mixin._createContentMarkup (node_modules/react/lib/ReactDOMComponent.js:289:32)
      at ReactDOMComponent.Mixin.mountComponent (node_modules/react/lib/ReactDOMComponent.js:199:12)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:38:35)
      at [object Object].ReactCompositeComponentMixin.mountComponent (node_modules/react/lib/ReactCompositeComponent.js:247:34)
      at [object Object].wrapper [as mountComponent] (node_modules/react/lib/ReactPerf.js:70:21)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:38:35)
      at [object Object].ReactCompositeComponentMixin.mountComponent (node_modules/react/lib/ReactCompositeComponent.js:247:34)
      at [object Object].wrapper [as mountComponent] (node_modules/react/lib/ReactPerf.js:70:21)
      at Object.ReactReconciler.mountComponent (node_modules/react/lib/ReactReconciler.js:38:35)
      at mountComponentIntoNode (node_modules/react/lib/ReactMount.js:248:32)
      at ReactReconcileTransaction.Mixin.perform (node_modules/react/lib/Transaction.js:134:20)
      at batchedMountComponentIntoNode (node_modules/react/lib/ReactMount.js:269:15)
      at ReactDefaultBatchingStrategyTransaction.Mixin.perform (node_modules/react/lib/Transaction.js:134:20)
      at Object.ReactDefaultBatchingStrategy.batchedUpdates (node_modules/react/lib/ReactDefaultBatchingStrategy.js:66:19)
      at Object.batchedUpdates (node_modules/react/lib/ReactUpdates.js:110:20)
      at Object.ReactMount._renderNewRootComponent (node_modules/react/lib/ReactMount.js:404:18)
      at Object.wrapper [as _renderNewRootComponent] (node_modules/react/lib/ReactPerf.js:70:21)
      at Object.ReactMount.render (node_modules/react/lib/ReactMount.js:493:32)
      at Object.wrapper [as render] (node_modules/react/lib/ReactPerf.js:70:21)
      at Object.ReactTestUtils.renderIntoDocument (node_modules/react/lib/ReactTestUtils.js:52:18)
```

**Comment by robcolburn** [comment]
I've meaning to do some deeper debugging. For now, (based on some console.log statements) what I know is:
* we get inside Slider's render function, and inside the conditional to create an InnerSlider
* we do not reach InnerSlider's render function

I haven't figured out to trigger node's repl debugger while running mocha. For now, reverting to 0.7 is working for me.

**Comment by akiran** [comment]
@robcolburn Are you seeing this issue with 0.8.1 as well ?
Last week end I released 0.8.2.  It may have problem but 0.8.1 should be fine

**Comment by robcolburn** [comment]
@akiran - started occurring on 0.8.1, attempted 0.8.2 and 0.8.3 before going back to 0.7.2

**Comment by jordansexton** [comment]
I was on 0.8.2 using React 0.13.3, and it was stable. I'm also running 0.7.2 now.

**Comment by mikecastro26** [comment]
Same issue happening here.

**Comment by gbrassey** [comment]
Same issue here with `react 0.13.3`. Had to roll back to `0.7.2`.

I get the issue with `0.8.0` and up.

**Comment by d-oliveros** [comment]
Same here

**Comment by rakeshok** [comment]
Same issue here

**Comment by jorgeartware** [comment]
yeah, me too :confused:

**Comment by archr** [comment]
+1

**Comment by akiran** [comment]
I will look into this today

**Comment by d-oliveros** [comment]
awesome, thanks

**Comment by rakeshok** [comment]
Counting on you :+1:

**Comment by jorgeartware** [comment]
sweet, can't wait :moyai:

**Comment by dsalehipour** [comment]
Awesome thank you, looking forward to the update!

**Comment by akiran** [comment]
react-slick@0.9.2 now works with both react 0.13, 0.14.
Please check and let me know if your issues are resolved

**Comment by gbrassey** [comment]
Still having trouble with react-slick@0.9.2 and react@0.13.3

```
Uncaught Error: Invariant Violation: Slider.render(): A valid ReactComponent must be returned. You may have returned undefined, an array or some other invalid object.
invariant @ invariant.js:42
ReactCompositeComponentMixin._renderValidatedComponent @ ReactCompositeComponent.js:821
wrapper @ ReactPerf.js:70
ReactCompositeComponentMixin.mountComponent @ ReactCompositeComponent.js:237
wrapper @ ReactPerf.js:70
ReactReconciler.mountComponent @ ReactReconciler.js:38
ReactMultiChild.Mixin.mountChildren @ ReactMultiChild.js:192
ReactDOMComponent.Mixin._createContentMarkup @ ReactDOMComponent.js:289
ReactDOMComponent.Mixin.mountComponent @ ReactDOMComponent.js:199
ReactReconciler.mountComponent @ ReactReconciler.js:38
ReactCompositeComponentMixin.mountComponent @ ReactCompositeComponent.js:247
wrapper @ ReactPerf.js:70ReactReconciler.mountComponent @ ReactReconciler.js:38
ReactCompositeComponentMixin.mountComponent @ ReactCompositeComponent.js:247
wrapper @ ReactPerf.js:70ReactReconciler.mountComponent @ ReactReconciler.js:38
mountComponentIntoNode @ ReactMount.js:248
Mixin.perform @ Transaction.js:134
batchedMountComponentIntoNode @ ReactMount.js:269
Mixin.perform @ Transaction.js:134
ReactDefaultBatchingStrategy.batchedUpdates @ ReactDefaultBatchingStrategy.js:66
batchedUpdates @ ReactUpdates.js:110
ReactMount._renderNewRootComponent @ ReactMount.js:404
wrapper @ ReactPerf.js:70
ReactMount.render @ ReactMount.js:493
wrapper @ ReactPerf.js:70
(anonymous function) @ homeEntry.js:7
```

**Comment by akiran** [comment]
@gbrassey  Can you post the example which is failing here ?
All demos worked well with react@0.13.3

**Comment by gbrassey** [comment]
I'm stumped. I can't get anything to work. Even something simple:
```
import React from 'react';
import Slider from 'react-slick';

var sliderMount = document.getElementById('slider-mount');
React.render(
  <Slider>
    <div>1</div>
  </Slider>
  , sliderMount);
```

from package.json:
```
  "dependencies": {
    "axios": "^0.5.4",
    "babel-core": "^5.6.15",
    "babel-loader": "^5.2.2",
    "json-loader": "^0.5.3",
    "node-libs-browser": "^0.5.2",
    "react": "^0.13.3",
    "react-slick": "^0.9.2",
    "webpack": "^1.10.1"
  },
```
Maybe webpack related?

**Comment by williamle8300** [comment]
@gbrassey are there any console errors?

**Comment by dheniges** [comment]
Confirming we are seeing the same issue. A coworker installed `0.9.2` last week and it runs fine, but installing it yesterday and today is throwing this error. 

Running `npm shrinkwrap` produces different outputs between the older and newer installs, with new installs showing the full react 0.14 dependency tree, so wondering if this is an issue with react 0.14?

**Comment by gbrassey** [comment]
react-slick@0.9.2 is including react@0.14.3 even though my project is using 0.13.3. The error is likely caused by two competing versions react. The project works if I delete `node_modules/react-slick/node_modules/react` (react-slick falls back to react@0.13.3).

**Comment by ChrisCinelli** [comment]
@gbrassey and @akiran : React should be a `peerDependency` (here they discuss the problem: https://github.com/yahoo/react-intl/issues/79 ). If you specify a react version in `devDependencies` or `dependencies`, react-slick will work only with that version. Check also https://medium.com/@dan_abramov/two-weird-tricks-that-fix-react-7cf9bbdef375#.3l87qpsws

**Comment by gbrassey** [comment]
thanks for the info and references @ChrisCinelli
